### PR TITLE
Move more of official build into build.ps1

### PR DESCRIPTION
**Pull request by jaredpar**
This change moves more of our core build logic into build.ps1. This continues to remove the line between our regular and official builds. Lets us more easily manage these assets as a single entity vs. having really two separate builds. 

Also adds more functionality to build.ps1

- pack: can create NuGet packages simply now
- sign: can run signing locally to verify changes

### CODE REVIEW

**Comment by jaredpar** [comment]
retest windows_release_unit32_prtest please

**Comment by jaredpar** [comment]
Looks like a flaky test with Microsoft.CodeAnalysis.Editor.UnitTests.Extensions.ISemanticSnapshotExtensionTests.TryGetSymbolTouchingPositionOnLeadingTrivia

**Comment by jaredpar** [comment]
Ping @dotnet/roslyn-infrastructure 

Need to get this in so I can move onto getting us setup in dotnet/versions

File: src/NuGet/BuildNuGets.csx [hunk 1]
```diff
@@ -280,17 +280,21 @@ int PackFiles(string[] nuspecFiles, string licenseUrl)
...
+        Console.WriteLine($"Packing {file}");
 
         p.Start();
         p.WaitForExit();
 
         var currentExit = p.ExitCode;
         if (currentExit != 0)
         {
+            Console.WriteLine($"nuget pack {p.StartInfo.Arguments}");
+            Console.WriteLine(p.StandardOutput.ReadToEnd());
```
Comment (jasonmalinowski) [inline src/NuGet/BuildNuGets.csx]: Isn't this now susceptible to the standard deadlock where the output stream can hit the full buffer size, and then .WaitForExit() blocks?

Comment (jasonmalinowski) [inline src/NuGet/BuildNuGets.csx]: (it looks like this was already broken for StandardError, but I suspect nuget pack is writing more to that then StandardError)

File: build/scripts/build.ps1 [hunk 2]
```diff
@@ -397,45 +516,38 @@ try {
...
-        exit 0
+    if ($sign) {
+        Run-SignTool
     }
 
-    if ($testPerfRun) {
-        Test-PerfRun
-        exit 0
+    # Must come after signing so that only the signed binaries are packed. Unlike 
+    # VSIX, NuGet doesn't support re-packing hence we have to order it this way.
```
Comment (jasonmalinowski) [inline build/scripts/build.ps1]: Comment seems odd: why is NuGet different than VSIX here? They're both OPC packages...?

File: src/NuGet/BuildNuGets.csx (hunk 1 above)
Comment (jaredpar) [inline src/NuGet/BuildNuGets.csx]: Possibly. But then hwat is the right way to read this output without barfing on the users screen?

File: build/scripts/build.ps1 (hunk 2 above)
Comment (jaredpar) [inline build/scripts/build.ps1]: VSIX is just a zip file. I can repack the contents slightly altered and all is well. 

NuGet is a zip file that also inserts undocumented (to my knowledge) files that need to be reproduced when content changes. Essentially it has files that are hashes of other files. If I knew how to reproduce those files then sure we could do the same thing here.

File: src/NuGet/BuildNuGets.csx (hunk 1 above)
Comment (jasonmalinowski) [inline src/NuGet/BuildNuGets.csx]: Standard workaround is you have to do the .ReadToEnd in a separate thread. Often easiest to just launch Task.Run() and do that. That ensures you have a thread reading the stream so it doesn't fill up. Then .Result the task once the .WaitForExit() returns.

File: src/NuGet/BuildNuGets.csx [hunk 3]
```diff
@@ -281,33 +281,17 @@ int PackFiles(string[] nuspecFiles, string licenseUrl)
...
-            }
-            else
-            {
-                message = $"{file}: error: {stdErr}";
-                ReportError(message);
-            }
-
-            File.AppendAllText(ErrorLogFile, Environment.NewLine + message);
+            Console.WriteLine($"nuget pack {p.StartInfo.Arguments}");
+            ReportError($"Pack operation failed with {currentExit}");
```
Comment (jaredpar) [inline src/NuGet/BuildNuGets.csx]: @jasonmalinowski removed all of the standard output / error redirection here. Turns out it was being done to not fail the script when we got the very specific error around building release packages with pre-release dependencies. That is actually no longer an error in Nuget 4.0 hence this is unnecessary

**Comment by 333fred** [comment]
LGTM, though I qualify that by saying that @tannergooding should probably look.

File: build/Targets/FixedPackages.props [hunk 4]
```diff
@@ -15,6 +15,7 @@
     <MicrosoftVisualStudioTextUIWpfFixedVersion>14.0.23205</MicrosoftVisualStudioTextUIWpfFixedVersion>
     <MicrosoftVisualStudioUtilitiesFixedVersion>14.0.23205</MicrosoftVisualStudioUtilitiesFixedVersion>
     <MicrosoftVSSDKBuildToolsFixedVersion>14.3.25420</MicrosoftVSSDKBuildToolsFixedVersion>
+    <MicrosoftVisualStudioComponentModelHostFixedVersion>14.0.25424</MicrosoftVisualStudioComponentModelHostFixedVersion>
```
Comment (tannergooding) [inline build/Targets/FixedPackages.props]: Why are we referencing the 14.0 version for all these?

File: build/scripts/build.ps1 [hunk 5]
```diff
@@ -375,14 +501,7 @@ try {
...
 
-    if ($msbuildDir -eq "") {
-        $msbuild = Ensure-MSBuild
-        $msbuildDir = Split-Path -parent $msbuild
-    }
-    else {
-        $msbuild = Join-Path $msbuildDir "msbuild.exe"
-    }
-
+    $msbuild, $msbuildDir = Ensure-MSBuildAndDir -msbuildDir $msbuildDir
```
Comment (tannergooding) [inline build/scripts/build.ps1]: I don't remember, which variable does this end up assigning?

**Comment by tannergooding** [comment]
A bit late, but 👍

File: build/Targets/FixedPackages.props (hunk 4 above)
Comment (jaredpar) [inline build/Targets/FixedPackages.props]: The Roslyn SDK still references them.

File: build/scripts/build.ps1 (hunk 5 above)
Comment (jaredpar) [inline build/scripts/build.ps1]: In this case both. The return of `Ensure-MSBUildAndDir` is a 2 element array and Powershell will splat assign to the variables. The real trivia is what happens when the array is length 1 ...

**Pull request merged**